### PR TITLE
Add OpenSSH key format support using PyCrypto

### LINKED ISSUE #3: Should be able to decrypt key on the fly
The library should be able to take the contents of an encrypted key and ask the user for the password when it needs to.

### PR DESCRIPTION
**Pull request by roaldnefs**
Add OpenSSH key format support when PyCrypto is installed. TransIP
will generate a key in the OpenSSH format.

After installing PyCrypto (pip install pycrypto) the following key
formats are supported:
- PKCS#1
- PKCS#8
- OpenSSH

This commit is backwards compatible with the old format (PKCS#1).

Closes #3

### CODE REVIEW

**Comment by benkonrath** [comment]
Can you also update the README and mention this is the preferred way to go? It's fine by me to only mention this method but I'm not sure if current users would find it confusing. Maybe we should start a changelog to list this change.

My big picture view on this feature: I would like remove old manual decryption style but this would be a breaking change and I'd want to do a major version bump. I'm thinking about cutting a 1.0 release soon. Removing the manual decryption would only happen in a 2.0 release.

**Comment by roaldnefs** [comment]
Updated the README.rst to mention PyCrypto.

PyCrypto is able to read the current format (PKCS#1), therefore the current users do not have to update their private key.

File: README.rst [hunk 1]
```diff
@@ -16,11 +16,11 @@ Prerequisite
...
-* Add your IP.
-* Generate key-pairs
-	+ Copy-paste the key into a file.
-	+ Reformat the key ``$ openssl rsa -in input.key -out decrypted_key``
-	+ put decrypted key next to this readme
+* Whitelist your IP.
+* Generate a new key-pair.
+	+ Copy-paste the private key into a file.
+	+ Either reformat the key to PKCS#1 format using ``$ openssl rsa -in input.key -out decrypted_key`` or install PyCrypto using ``$ pip install pycrypto``.
+	+ Put the private key in a file called ``decrypted_key`` besides this ``README.rst`` file.
```
Comment (benkonrath) [inline README.rst]: Small typo: besides -> beside

**Comment by benkonrath** [comment]
Thanks for the update. I'll merge it when the typo is fixed.

**Comment by benkonrath** [comment]
Just saw this:
> PyCrypto is able to read the current format (PKCS#1), therefore the current users do not have to update their private key.

Yeah, exactly. The way you've developed it now is great. Hopefully I can phase out and remove the support for the decrypted key.

**Pull request merged**